app/usecases/users.py

```diff
@@ -192,3 +200,142 @@ async def update_email_address(
 
 async def fetch_total_registered_user_count() -> int:
     return await users.fetch_total_registered_user_count()
+
+
+async def delete_one_by_user_id(user_id: int, /) -> None | Error:
+    """\
+    An anonymization process for user deletion, mainly implemented
+    for the purpose of complying with GDPR, CCPA and other regulations.
+    """
```

Comment:
There are still quite a few open questions for "full compliance", but this is a pretty good start